src/understand-tidb/cbo.md

```diff
@@ -1 +1,165 @@
+
+1. The `getTask` and `storeTask` functions are used for memory search. The `getTask` function is used to find the optimal physical plan from the hash table when the corresponding path has been calculated.  And the `storeTask` function is used to store the best physical plan which meets the required properties.
+2. `p.self.exhaustPhysicalPlans(prop)`: Generates all possible plans that can match the required property. We will introduce this part in detail in the later `Exhaust Physical Plans` section.
+3. `p.enumeratePhysicalPlans4Task(plansFitsProp, prop)`: Select the lowest cost plan among all generated physical plans. We will introduce this part in detail in the later `Enumerate Physical Plans For Task` section.
+
+In addition to `baseLogicalPlan.findBestTask`, we also need to focus on `DataSource.findBestTask`. It is usually the end of the recursive call of the `findBestTask` interface function. It will enumerate all the available paths and choose a plan with least cost. However, due to space limitations, the implementation of `findBestTask` for other operators will not be introduced in more detail. You can read the code to learn more.
+
+We will illustrate how the `baseLogicalPlan.findBestTask` work with the following example:
+
+```sql
+select sum(s.a),count(t.b) from s join t on s.a = t.a and s.c < 100 and t.c > 10 group by s.a; // Note: Assume that we have index(s.a) and index(t.b).
+```
+
+The logical plan corresponding to this SQL is as follows:
+
+![logical-plan](../img/cbo-example-logicalPlan.png)
+
+Then we will combine the following picture and code for a detailed introduction to explain how the `baseLogicalPlan.findBestTask` function work. 
+
+In the figure, the black font operator is a logical operator(e.g, `Agg`, `Join` and `DS`), the blue font is a physical operator(e.g, `Stream Agg`, `Hash Join` and `Index Scan(a)`...), and the yellow arrow is an operator for which the cost has been calculated. And the red dashed arrow is not in compliance with `prop` operator. The font on the arrow describes the property requirements for the lower operators(e.g, `s.a` means the the output of the lower operator needs to be ordered according to `s.a`).
+
+1. `Step1`: The `getTask` function corresponds to the yellow arrow in the figure, which means that the calculated part can be used directly without repeated calculations.
+2. `Step2`: The `p.self.exhaustPhysicalPlans(prop)` function in the figure represents the process of logical operator generating physical operator. For example, logical operator `Agg` generates physical operators `Stream Agg` and `Hash Agg`. But some physical operators that do not satisfy the required property cannot be generated. For example, if the required property is `s.a` logical operator `DS`, only physical operator `Index Scan(a)` can be generated, but `TableScan` cannot be generated.
```

Comment:
I don't get this part:

> The `p.self.exhaustPhysicalPlans(prop)` function in the figure represents the process of logical operator generating physical operator.

I can't see `p.self.exhaustPhysicalPlans(prop)` in any figure. Do you mean the figure actually explains what `p.self.exhaustPhysicalPlans(prop)` does or do I miss something?